crates/uv-cache/src/lib.rs

```
@@ -814,6 +780,48 @@ impl ArchiveTimestamp {
         Ok(Self::Exact(Timestamp::from_metadata(&metadata)))
     }
 
+    /// Return the modification timestamp for a source tree, i.e., a directory.
+    ///
+    /// If the source tree doesn't contain an entrypoint (i.e., no `pyproject.toml`, `setup.py`, or
+    /// `setup.cfg`), returns `None`.
+    pub fn from_source_tree(path: impl AsRef<Path>) -> Result<Option<Self>, io::Error> {
+        // Compute the modification timestamp for the `pyproject.toml`, `setup.py`, and
+        // `setup.cfg` files, if they exist.
+        let pyproject_toml = path
+            .as_ref()
+            .join("pyproject.toml")
+            .metadata()
+            .ok()
```

Should we raise those if they aren't file not found errors?